### PR TITLE
[JBTM-3109] LRA TCK Arquillian rework

### PR DESCRIPTION
**Pull request by ochaloup**
https://issues.jboss.org/browse/JBTM-3109
https://issues.jboss.org/browse/JBTM-3113

Refactoring of the LRA Client to comply with LRA spec running TCKs with Arquillian.

LRA spec work:
https://github.com/eclipse/microprofile-lra/issues/94
https://github.com/eclipse/microprofile-lra/pull/87
https://github.com/eclipse/microprofile-lra/pull/96

!QA_JTA !QA_JTS_JDKORB !QA_JTS_OPENJDKORB !QA_JTS_JACORB !BLACKTIE !XTS !PERF NO_WIN !AS_TESTS !TOMCAT !JACOCO !mysql !postgres !db2 !oracle !RTS

### CODE REVIEW

File: rts/lra/lra-client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java [hunk 1]
```diff
@@ -909,12 +895,12 @@ private boolean isStatus(URL lraId, CompensatorStatus testStatus) {
...
      */
-    private static Optional<CompensatorStatus> fromString(String status) {
+    private static LRAStatus fromString(String status) {
         if (status != null) {
-            return Optional.of(CompensatorStatus.valueOf(status));
+            return LRAStatus.valueOf(status);
         }
 
-        return Optional.empty();
+        return null;
```
Comment (mmusgrov) [inline rts/lra/lra-client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java]: The status should never be null, instead it should either throw generic exception (including NotFound if that is what the response from the coordinator indicated).

File: rts/lra/lra-proxy/api/src/main/java/io/narayana/lra/client/internal/proxy/ParticipantProxy.java [hunk 2]
```diff
@@ -88,19 +88,19 @@ void setFuture(Future<Void> future, boolean compensate) {
         this.compensate = compensate;
     }
 
-    private CompensatorStatus getExpectedStatus() {
-        return compensate ? CompensatorStatus.Compensated : CompensatorStatus.Completed;
+    private LRAStatus getExpectedStatus() {
```
Comment (mmusgrov) [inline rts/lra/lra-proxy/api/src/main/java/io/narayana/lra/client/internal/proxy/ParticipantProxy.java]: We now have separate status types for participants and LRAs (ParticipantStatus and LRAStatus). So the return here should be ParticipantStatus. There are lots of other places in the PR where we need to be returning ParticipantStatus and not LRAStatus.

File: rts/lra/lra-client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java (hunk 1 above)
Comment (ochaloup) [reply rts/lra/lra-client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java]: ok, agree. I think the method `fromString(...)` should throw the `IllegalArgumentException` as defined in its javadoc

File: rts/lra/lra-proxy/api/src/main/java/io/narayana/lra/client/internal/proxy/ParticipantProxy.java (hunk 2 above)
Comment (ochaloup) [reply rts/lra/lra-proxy/api/src/main/java/io/narayana/lra/client/internal/proxy/ParticipantProxy.java]: thanks! this was pretty wrong

**Comment by ochaloup** [comment]
@mmusgrov the TCK tests should be passing now

**Comment by ochaloup** [comment]
@tomjenkinson the jira was created and the commit description was updated

**Comment by mmusgrov** [comment]
Please ignore the MAIN and BLACKTIE profile tests. I was setting up a PR poller job on our Jenkins installation. I temporarily added a build periodically schedule and it started retesting all the PRs. Since the job config was incorrect it rapidly cycled through old PRs before I was able to stop it. Sorry about that. It is disabled now.

**Pull request merged**